File: Ravi/Microsoft/Day 1/EvaluateReversePolishNotation.java
class Solution {
    public int evalRPN(String[] tokens) {
        
        Stack<Integer> st = new Stack<>();
        
        for(String s : tokens) {
            try {
                
                // It will return the integer if possible or will sent the flow to catch block
                int a = Integer.parseInt(s);
                st.push(a);
            } catch(Exception e) {
                
                // We will pop the first two integers from the stack
                int a = st.pop();
                int b = st.pop();
                
                // We will check which operation we need to perform on the operands and will perform those pperations
                if(s.charAt(0) == '+') st.push(a + b);
                else if(s.charAt(0) == '-') st.push(b - a);
                else if(s.charAt(0) == '*') st.push(b * a);
                else st.push(b / a);
                
            }
        }
        
        return st.pop();
    }
}

File: Sahil726/Microsoft/Evaluate Reverse Polish Notation.cpp
long long perform(char ch,long long  a,long long  b){
    if(ch=='+') return a+b;
    else if(ch=='*') return a*b;
    else if(ch=='-') return b-a;
    else return b/a;
}
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
    stack<long long>st;
    for(int i=0;i<tokens.size();i++){
        if(tokens[i]=="+" || tokens[i]=="*" || tokens[i]=="-" || tokens[i]=="/"){
        long long a=st.top();st.pop();
        long long b=st.top();st.pop();
        st.push(perform(tokens[i][0],a,b));
        }
        else{
        long long  n=stoi(tokens[i]);
        st.push(n);
        }
    }
    return st.top();
    }
};
